trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesTraitsService.php

```diff
@@ -414,4 +418,97 @@ public function getMethodUnitDataType($unit_id) {
 
     return $data_type ?? 0;
   }
-}
+
+  /**
+   * Get trait, method and unit combination.
+   * 
+   * @param string|int $trait
+   *   A string value is the trait name, whereas an integer value is the trait id number.
+   * @param string|int $method
+   *   A string value is the trait method short name, whereas an integer value is the method id number.
+   * @param string|int $unit
+   *   A string value is the method unit name, whereas an integer value is the unit id number.
+   * 
+   * @return array
+   *   An associative array where the keys are trait, method and unit and the values
+   *   are the cvterm records for each key.
+   * 
+   *   0 if any one of trait, method or unit did not return any record.
+   * 
+   * @dependencies
+   *   getTrait(), getTraitMethod(), getMethodUnit() and getMethodUnitDataType().
+   */
+  public function getTraitMethodUnitCombo(string|int $trait, string|int $method, string|int $unit) {
```

Comment:
These parameters which are set up to take 2 different types is great! As we discussed in the office, it'd be really helpful to have the individual getter methods take these parameters as well. So:

- `getTrait(string|int $trait)`
- `getTraitMethod(string|int $trait)`
- `getMethodUnit(string|int $method)`

This way, `getTraitMethodUnitCombo` doesn't need to deal with creating an array to pass into each of the methods, or detecting int vs string, and all the getters are kept consistent.

Reply:
This will require a major overhaul of the service and not to mention the need to revise all reference to each method in many places to comply with the new parameter setup. I will start a separate issue that will update this service that includes the limit clause pointed out. in the meantime, if you accept this method, your source code will not be affected by update.

Reply:
I don't think this needs to be done in a different PR. Especially as it is only these 3 getters we are suggesting to change right now and they are only used within the service and the ValidTraitTest so it will not affect as much as you worry 💞